src/emulator/functionsEmulatorShared.ts

```diff
@@ -128,6 +129,33 @@ export class EmulatedTrigger {
   }
 }
 
+/**
+ * Creates a unique trigger definition for each region a function is defined in.
+ * @param definitions A list of all CloudFunctions in the deployment.
+ * @return A list of all CloudFunctions in the deployment, with copies for each region.
+ */
+export function emulatedFunctionsByRegion(
+  definitions: EmulatedTriggerDefinition[]
+): EmulatedTriggerDefinition[] {
+  const regionDefinitions: EmulatedTriggerDefinition[] = [];
+  for (const def of definitions) {
+    if (!def.regions) {
+      def.regions = ["us-central1"];
+    }
+    // Create a separate CloudFunction for
+    // each region we deploy a function to
+    for (const region of def.regions) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const defDeepCopy: EmulatedTriggerDefinition = JSON.parse(JSON.stringify(def));
+      defDeepCopy.regions = [region];
+      defDeepCopy.id = `${region}-${defDeepCopy.name}`;
```

Comment:
So it seems like functions never have an `id` before this function is run, but afterwards they all have an `id`. Could we maybe improve typing of input/output to make the compiler know this?

Reply:
Done